dags/airflow_db_prune_logs_rotate.py

```diff
@@ -0,0 +1,140 @@
+    "depends_on_past": False,
+    "start_date": pendulum.datetime(2015, 12, 1, tz="America/Chicago"),
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 0,
+    "on_failure_callback": task_fail_slack_alert,
+}
+
+
+@task(task_id="get_days_back_to_prune")
+def get_days_back_to_prune(days_back_to_prune: int):
+    """
+    Retrieve the number of days to prune back from the Airflow database.
+
+    Args:
+        days_back_to_prune (int): Number of days to look back for pruning.
+
+    Returns:
+        int: The number of days to prune back.
+    """
+    prune_before_days = int(days_back_to_prune)
+    logger = LoggingMixin().log
+    logger.info(f"Pruning Airflow DB records older than {prune_before_days} days")
+    return prune_before_days
+
+
+@task(task_id="get_clean_before_timestamp")
+def get_clean_before_timestamp(prune_before_days: int):
+    """
+    Calculate the ISO8601 timestamp before which records should be pruned.
+
+    Args:
+        prune_before_days (int): Number of days to look back for pruning.
+
+    Returns:
+        str: ISO8601 formatted timestamp.
+    """
+    clean_before_timestamp = (
+        pendulum.now("America/Chicago") - pendulum.duration(days=prune_before_days)
+    ).to_iso8601_string()
+    logger = LoggingMixin().log
+    logger.info(
+        f"Pruning records before {pendulum.parse(clean_before_timestamp).to_datetime_string()} America/Chicago, ISO8601: {clean_before_timestamp}"
+    )
+    return clean_before_timestamp
+
+
+@task.bash(task_id="airflow_db_clean")
+def db_clean_bash(timestamp: str) -> str:
+    """
+    Generate the bash command to clean the Airflow database before a given timestamp.
+
+    Args:
+        timestamp (str): ISO8601 formatted timestamp.
+
+    Returns:
+        str: Bash command string.
+    """
+    cmd = f'airflow db clean --yes --clean-before-timestamp "{timestamp}"'
+    logger = LoggingMixin().log
+    logger.info(f"Running command: {cmd}")
+    return cmd
+
+
+@task.bash(task_id="airflow_log_file_cleanup")
+def log_file_cleanup_bash(day_interval: int) -> str:
+    """
+    Generate the bash command to delete Airflow log files older than a given number of days.
+
+    Args:
+        day_interval (int): Number of days; logs older than this will be deleted.
+
+    Returns:
+        str: Bash command string.
+    """
+    cmd = (
+        f'find /opt/airflow/logs/ -type f -name "*.log" -mtime +{day_interval} -delete'
+    )
+    logger = LoggingMixin().log
+    logger.info(f"Running command: {cmd}")
+    return cmd
+
+
+@task.bash(task_id="airflow_log_dir_cleanup")
+def log_dir_cleanup_bash() -> str:
+    """
+    Generate the bash command to delete empty directories in the Airflow logs directory.
+
+    Returns:
+        str: Bash command string.
+    """
+    cmd = "find /opt/airflow/logs/ -type d -empty -delete"
+    logger = LoggingMixin().log
+    logger.info(f"Running command: {cmd}")
+    return cmd
+
+
+@dag(
```

Comment:
you asked wether or not I think we should set a max duration for this DAG, and I think it would be good to do that and it set it to a value that would be cause for concern. an hour?

Reply:
I think setting to a shorter limit is good. If we somehow accumulate more logs that can be run on our regular schedule, that's probably a problem that will take fixing outside of this dag 🤷

Reply:
https://github.com/cityofaustin/atd-airflow/pull/285/commits/0e5e2708e84cf368c3ef3cbf16f3fc002e28bc64

Bummer, I didn't see a obvious method to control Airflow from the DAG that would allow the `dagrun_timeout` to be set conditionally, so I set it at 10 minutes. I agree, a shorter interval is better, and we can adjust once we get more experience with it.

Thank you for your suggestion 🙏